package/tests/test_integrations/test_sqlite_store.py

```diff
@@ -29,13 +29,145 @@ def db_session_class(store_path):
+    session.add(RunModel(id="1", blob="blob1"))
+    session.commit()
+    session.close()
+    yield Path(database_loc)
+
+
+@pytest.fixture
+def mock_db2(tmp_path):
+    database_loc = str(tmp_path / "db2.db")
+    engine = create_engine(f"sqlite:///{database_loc}")
+    session_class = sessionmaker(bind=engine)
+    Base.metadata.create_all(bind=engine)
+    session = session_class()
+    session.add(RunModel(id="2", blob="blob2"))
+    session.commit()
+    session.close()
+    yield Path(database_loc)
+
+
+@pytest.fixture
+def mock_db3_with_db2_data(tmp_path):
+    database_loc = str(tmp_path / "db3.db")
+    engine = create_engine(f"sqlite:///{database_loc}")
+    session_class = sessionmaker(bind=engine)
+    Base.metadata.create_all(bind=engine)
+    session = session_class()
+    session.add(RunModel(id="2", blob="blob2"))
+    session.commit()
+    session.close()
+    yield Path(database_loc)
+
+
+def get_files_in_bucket(bucket_name):
+    s3 = boto3.client("s3")
+    response = s3.list_objects(Bucket=bucket_name)
+    files = [obj["Key"] for obj in response.get("Contents", [])]
+    return files
+
+
+@pytest.fixture
+def mocked_db_in_s3(mocked_s3_bucket, mock_db1, mock_db2):
+    # define the name of the S3 bucket and the database file names
+    db1_filename = "db1.db"
+    db2_filename = "db2.db"
+
+    # upload each mock database file to the mocked S3 bucket
+    mocked_s3_bucket.put_object(
+        Bucket=BUCKET_NAME, Key=db1_filename, Body=mock_db1.read_bytes()
+    )
+    mocked_s3_bucket.put_object(
+        Bucket=BUCKET_NAME, Key=db2_filename, Body=mock_db2.read_bytes()
+    )
+
+    return get_files_in_bucket(BUCKET_NAME)
+
+
+@pytest.fixture
+def mocked_db_in_s3_repeated_runs(
+    mocked_s3_bucket, mock_db1, mock_db2, mock_db3_with_db2_data
+):
+    # define the name of the S3 bucket and the database file names
+    db1_filename = "db1.db"
+    db2_filename = "db2.db"
+    db3_filename = "db3.db"
+
+    # upload each mock database file to the mocked S3 bucket
+    mocked_s3_bucket.put_object(
+        Bucket=BUCKET_NAME, Key=db1_filename, Body=mock_db1.read_bytes()
+    )
+    mocked_s3_bucket.put_object(
+        Bucket=BUCKET_NAME, Key=db2_filename, Body=mock_db2.read_bytes()
+    )
+    mocked_s3_bucket.put_object(
+        Bucket=BUCKET_NAME, Key=db3_filename, Body=mock_db3_with_db2_data.read_bytes()
+    )
+
+    return get_files_in_bucket(BUCKET_NAME)
+
+
+def session_id():
+    i = 0
+    while True:
+        yield f"session_{i}"
+        i += 1
+
+
+def test_get_dbname_with_env_var(mocker):
+    mocker.patch.dict(os.environ, {"KEDRO_SQLITE_STORE_USERNAME": "env_user_name"})
+    mocker.patch("getpass.getuser", return_value="computer_user_name")
+    dbname = _get_dbname()
+    assert dbname == "env_user_name.db"
+
+
+def test_get_dbname_without_env_var(mocker):
+    mocker.patch(
+        "os.environ",
+        cast(Dict[str, str], {**os.environ, "KEDRO_SQLITE_STORE_USERNAME": None}),
+    )
```

Comment:
```suggestion
```

I guess there's a good reason for this line to exist but I don't see why?

Reply:
So if I don't mock the environment variable explicitly to None it takes the previous tests environment variable. Rearranging tests also did not help because it still took the environment variable incase your environment had one.

Mocking it without cast threw one of the two (i can't remember): mypy or pylint error so with this the linting passes. :\

Reply:
I think mocking like this should fix it: https://stackoverflow.com/a/31583528. We should definitely have this set up so that the mocking of environment variables doesn't leak from one test into another.

If you still have a linting error after trying that then I'd just put in an `ignore` rather than `cast`.

Reply:
it works as you suggested. also makes it easier to revert to a one line return in the sqlite_store